### PR TITLE
Removed DatabaseIntrospection.get_key_columns().

### PR DESCRIPTION
**Pull request by felixxm**
Thanks Simon Charette for the report.

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/mysql/base.py [hunk 1]
```diff
@@ -308,8 +308,9 @@ def check_constraints(self, table_names=None):
                 primary_key_column_name = self.introspection.get_primary_key_column(cursor, table_name)
                 if not primary_key_column_name:
                     continue
-                key_columns = self.introspection.get_key_columns(cursor, table_name)
-                for column_name, referenced_table_name, referenced_column_name in key_columns:
+                relations = self.introspection.get_relations(cursor, table_name)
```
Comment (timgraham) [inline django/db/backends/mysql/base.py:311]: Did you consider?
`for column_name, (referenced_column_name, referenced_table_name) in relations.items():`
(and similar for sqlite3)

File: docs/releases/4.1.txt [hunk 2]
```diff
@@ -246,6 +246,9 @@ backends.
 * ``BaseDatabaseFeatures.has_case_insensitive_like`` is changed from ``True``
   to ``False`` to reflect the behavior of most databases.
 
+* ``BaseDatabaseIntrospection.get_key_columns()`` is removed, use
```
Comment (timgraham) [inline docs/releases/4.1.txt:249]: , -> .

File: docs/releases/4.1.txt [hunk 3]
```diff
@@ -246,6 +246,9 @@ backends.
...
+  ``BaseDatabaseIntrospection.get_relations()`` instead.
```
Comment (timgraham) [inline docs/releases/4.1.txt:250]: Perhaps these should just be "DatabaseIntrospection". The "Base" implementations can't be used anyway.

**Review by timgraham — APPROVED** [review]

**Review by charettes — APPROVED** [review]
Thanks for the adjustments @felixxm

**Comment by felixxm** [comment]
Thanks both for reviews :1st_place_medal:

File: django/db/backends/mysql/base.py [hunk 4]
```diff
@@ -308,8 +308,7 @@ def check_constraints(self, table_names=None):
                 primary_key_column_name = self.introspection.get_primary_key_column(cursor, table_name)
                 if not primary_key_column_name:
                     continue
-                key_columns = self.introspection.get_key_columns(cursor, table_name)
```
[the remark this answers is not in the document]
Comment (felixxm) [reply django/db/backends/mysql/base.py:311]: fixed :facepalm:

**Pull request merged**